src/hoplon/core.cljs

```diff
@@ -414,35 +500,31 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ;; HTML Constructors ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-(defn- mksingleton [elem]
+(defn- mksingleton [tag]
   "Retrieves the DOM element `elem` from js/document and updates in-place.
+
   Creates the element if missing."
   (fn [& args]
-   (let [oelem (obj/get js/document elem)]
-     (when-not oelem
-       (obj/set js/document elem
-         (.createElement js/document elem)))
-     (with-let [helem (->hoplon oelem)]
-       (let [[attrs kids] (parse-args args)]
-         (when-not (:static attrs)
-           (merge-kids helem nil nil)
-           (add-attributes! helem attrs)
-           (add-children! helem kids)))))))
+    (if-let [elem (obj/get js/document tag)]
+      (-elem! elem :hoplon/invoke args)
+      (with-let [elem (.createElement js/document tag)]
+        (obj/set js/document tag elem)
+        (-elem! elem :hoplon/invoke args)))))
```

Comment:
this could potentially be simplified even further with a recursive call

Reply:
^^ is a bit wrong because it doesn't consider the inner fn, but you get the idea...